packages/core/data-transfer/lib/providers/local-strapi-destination-provider/index.ts

```diff
@@ -95,4 +96,25 @@ class LocalStrapiDestinationProvider implements IDestinationProvider {
 
     return mapSchemasValues(schemas);
   }
+
+  getAssetsStream(): NodeJS.WritableStream {
+    if (!this.strapi) {
+      throw new Error('Not able to stream Assets. Strapi instance not found');
+    }
+
+    const assetsDirectory = path.join(this.strapi.dirs.static.public, 'uploads');
+
+    return new Writable({
+      objectMode: true,
+      async write(chunk, _encoding, callback) {
+        const entryPath = path.join(assetsDirectory, chunk.file);
+        const writableStream = fse.createWriteStream(entryPath);
```

Comment:
We should have some logic here to check that we're doing a "restore" and if so, delete the uploads folder (or something like rename existing uploads -> unpack these files to new empty uploads -> delete the original so that we have a rollback option).  Merge doesn't need to be implemented yet, but it will need to check if files already exist and either overwrite them, skip, or abort.

Reply:
I would recommend doing something like an after transfer for cleanup if that's the case for rollbacks, incase of success of the whole transfer we delete the old uploads

Reply:
Yeah, I think we should do something like:
- rename "uploads" -> "uploads_backup_{timestamp}"
- restore to "uploads"
  - on success, delete "uploads_backup_{timestamp}"
  - on failure
    - delete "uploads"
    - rename "uploads_backup_{timestamp}" -> "uploads"
      - on failure of rollback, print a message informing that original files are in "uploads_backup_{timestamp}" but that we failed to restore them back to "uploads"